### PR TITLE
Remove COS integration test

### PR DESCRIPTION
**Pull request by amandahla**
### Overview

This PR remove COS during the integration tests.

### Rationale

Since COS integration is not required for Synapse to work, we decided to remove it from the integration tests to make them faster. 

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Manging Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`
- [x] The documentation for charmhub is updated.
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

File: .github/workflows/integration_test.yaml [hunk 1]
```diff
@@ -8,7 +8,7 @@ jobs:
     uses: canonical/operator-workflows/.github/workflows/integration_test.yaml@main
     secrets: inherit
     with:
-      extra-arguments: --localstack-address 172.17.0.1
+      extra-arguments: --localstack-address 172.17.0.1 -m "not cos"
```
Comment (cbartz) [inline .github/workflows/integration_test.yaml:11]: Would it make more sense to remove these tests? If they're not run, they'll probably stop working after a while. As they are checked in version control, they can be restored later if you wish.

Comment (amandahla) [reply .github/workflows/integration_test.yaml:11]: We are planning to run them before publishing to edge :)

Comment (cbartz) [reply .github/workflows/integration_test.yaml:11]: Gotcha, could you please add this to the PR description?

**Review by cbartz — APPROVED** [review]

File: .github/workflows/integration_test.yaml (hunk 1 above)
Comment (amandahla) [reply .github/workflows/integration_test.yaml:11]: Sure, thanks for raising this.

**Comment by nrobinaubertin** [comment]
How will we test them before pushing to edge ? Will there be an action ?

**Comment by amandahla** [comment]
> How will we test them before pushing to edge ? Will there be an action ?

I was planning to create a branch in operator-workflows and change the publish charm by adding a step to run skipped integration tests but for this COS case I'm very tempted to just remove them. What do you think?

**Comment by nrobinaubertin** [comment]
> > How will we test them before pushing to edge ? Will there be an action ?
> 
> I was planning to create a branch in operator-workflows and change the publish charm by adding a step to run skipped integration tests but for this COS case I'm very tempted to just remove them. What do you think?

I actually agree, the COS tests that we currently have don't seem to really test the synapse charm per se. Those tests could be only done in staging

**Comment by javierdelapuente** [comment]
> > How will we test them before pushing to edge ? Will there be an action ?
> 
> I was planning to create a branch in operator-workflows and change the publish charm by adding a step to run skipped integration tests but for this COS case I'm very tempted to just remove them. What do you think?

I am not sure we should test at that point. I would rather delete it (of leave the test where it is). At that point the code is in main, and we will have broken CI/CD there that will require another PR....

**Comment by jdkandersson** [comment]
> > > How will we test them before pushing to edge ? Will there be an action ?
> > 
> > 
> > I was planning to create a branch in operator-workflows and change the publish charm by adding a step to run skipped integration tests but for this COS case I'm very tempted to just remove them. What do you think?
> 
> I am not sure we should test at that point. I would rather delete it (of leave the test where it is). At that point the code is in main, and we will have broken CI/CD there that will require another PR....

@merkata did we discuss writing a spec for e2e tests for how to do COS testing? the comment from @javierdelapuente is quite reasonable that we should not have a test that isn't run

**Review by gtrkiller — APPROVED** [review]

**Review by cbartz — APPROVED** [review]

**Pull request merged**